File: FEBRUARY/16-02-2026/Meeting Rooms.cpp
class Solution {
  public:
    static bool cmp(vector<int>& first,vector<int>& second){
        if(first[1] == second[1]) return first[0]<=second[0];
        return first[1]<second[1];
    }
    bool canAttend(vector<vector<int>> &arr) {
        
        // inshort there sould be no any overlap b/w any two intervals
        sort(arr.begin(),arr.end(),cmp);
        
        int n = arr.size();
        if(n == 1) return true;
        
        for(int i = 1;i<n;i++){
            int prevEnd = arr[i-1][1];
            int currentStart = arr[i][0];
            if(prevEnd>currentStart) return false;
        }
        
        return true;
        
    }
};

File: 2026_GFG_POTD/FEBRUARY/16-02-2026/Meeting Rooms.java
class Solution {
    static boolean canAttend(int[][] arr) {
        // code here
        Arrays.sort(arr,(a,b)->a[0]-b[0]);
        int n=arr.length;
        
        for(int i=0;i<n-1;i++)
        {
            int nextStart=arr[i+1][0];
            int currEnd=arr[i][1];
            if(currEnd>nextStart)
            {
                return false;
            }
        }
        
        return true;
    }
}